core/src/main/java/org/elasticsearch/discovery/Discovery.java

```diff
@@ -87,4 +88,10 @@ public FailedToCommitClusterStateException(String msg, Throwable cause, Object..
             super(msg, cause, args);
         }
     }
+
+    /**
+     * @return stats about the discovery
+     */
+    PendingClusterStateStats stats();
```

Comment:
can we make this return a DiscoveryStats interface which extends both toXContent and Streamable? ZenDiscovery can return a ZenDiscoveryStats object and other discoveries can decide what they want to expose. This also means we need a method `readStatsFrom(InputStream)` to read a DiscoveryStats object from a stream (only the discovery implementation knows how to deserialize it).